src/test/scala/ValidatorTest.scala

```diff
@@ -62,4 +63,47 @@ class ValidatorTest extends FunSuite {
     assert(error.column === "2")
     assert(error.content === "Age")
   }
+
+  test("should set error when primary keys are not unique") {
+    val uri = new URI(
+      s"file://${new File(s"${csvwExamplesBaseDir}observations_duplicate_primary_key.csv-metadata.json").getAbsolutePath}"
+    )
+    val validator = new Validator(uri)
+    validator.validate()
+    assert(validator.errors.length === 1)
+    val error = validator.errors(0)
+    assert(error.`type` === "duplicate_key")
+    assert(
+      error.content === "key already present - W00000001,owned-owned-with-a-mortgage-or-loan-or-shared-ownership,Y0T15,very-good-or-good-health"
+    )
+    assert(error.category === "schema")
+  }
+
+  test(
+    "it should NOT set primary key violation if datetime value is equal in UTC and the strings representing them differ"
```

Comment:
Maybe `... in UTC and the timezones differ`.

Reply:
👍